File: backend/solana_util.js
// Solana NFT object.
// TODO: This class needs to be finalized.
class SolanaNFT {
	// Construct an NFT from an image URL.  
	constructor(imgUrl) {
		this.imgUrl = imgUrl;
	}

	// Return the image URL associated with the NFT.
	getImgUrl() {
		return this.imgUrl;
	}
}

// Request for a query of the NFTs owned by a wallet.
class SearchRequest {
	// Construct a request containing a wallet.
	constructor(wallet) {
		this.wallet = wallet;
	}
}

// Response from a query of the NFTs owned by a wallet.
class SearchResponse {
	// Construct a response containing a list of NFT objects.
	constructor(nftList) {
		this.nftList = nftList;
	}
}

module.exports = { SolanaNFT, SearchRequest, SearchResponse, };

File: backend/solana_util.py
import json

#-----------------------------------------------------------------------

# Solana NFT object.
class SolanaNFT:
	# Constructs an NFT from an image URL.
	def __init__(self, address, metadata, new=False):
		self.address = address
		self.metadata = metadata

	# Returns the image URL associated with the NFT.
	def get_img_url(self):
		return self.metadata["image"]

#-----------------------------------------------------------------------

# Request for a query of the NFTs owned by a wallet.
class SearchRequest:
	# Constructs a request containing a wallet.
	def __init__(self, wallet):
		self.wallet = wallet

#-----------------------------------------------------------------------

# Response from a query of the NFTs owned by a wallet.
class SearchResponse:
	# Constructs a response containing a list of NFT objects.
	def __init__(self, nft_list):
		self.nft_list = nft_list

	# Serializes object to a JSON formatted string.
	def json(self):
		return json.dumps(self, default=lambda o: o.__dict__)